### PR TITLE
feat: add UTXO release mechanism for nano contract transactions

### PR DESCRIPTION
**Pull request by raul-oliveira**
### Acceptance criteria
  - Add TTL (60s) to markUtxoSelected calls in NanoContractTransactionBuilder as safety net against permanent UTXO locks
  - Add releaseUtxos() to ISendTransaction interface and SendTransaction class for explicit UTXO cleanup on rejection/abandonment
  - Add no-op releaseUtxos() to SendTransactionWalletService
  - Add unit tests for releaseUtxos (happy path, null tx, null storage, mid-loop error)
  - Add integration test validating lock/unlock lifecycle: prepare -> fail ->release -> succeed
  
### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

## Summary by CodeRabbit

* **New Features**
  * Users can now release previously locked UTXOs from prepared transactions, improving flexibility when reusing funds.

* **API**
  * Send-transaction interfaces expose a release operation so clients can trigger UTXO release when needed.

* **Tests**
  * Added unit and integration tests covering the UTXO lock/unlock lifecycle and release behavior to ensure robustness.

### CODE REVIEW

File: __tests__/integration/fullnode-specific/nano_utxo_lock_unlock.test.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  stopAllWallets,
+  waitForTxReceived,
+  waitTxConfirmed,
+} from './helpers/wallet.helper';
+import HathorWallet from '../../src/new/wallet';
+import { NATIVE_TOKEN_UID, NANO_CONTRACTS_INITIALIZE_METHOD } from '../../src/constants';
+import { NanoContractTransactionError } from '../../src/errors';
+import { SendTransaction } from '../../src';
+
+describe('Nano contract UTXO lock/unlock lifecycle', () => {
```
Comment (tuliomir) [inline __tests__/integration/fullnode-specific/nano_utxo_lock_unlock.test.ts:21]: chore: Please move this test suite to the fullnode-specific tests folder: `__tests__/integration/fullnode-specific/nano_utxo_lock_unlock.test.ts`.

This root `integration` folder is being discontinued.

File: src/new/sendTransaction.ts [hunk 2]
```diff
@@ -687,6 +687,32 @@ export default class SendTransaction extends EventEmitter implements ISendTransa
...
+    for (const input of this.transaction.inputs) {
+      try {
+        await this.storage.utxoSelectAsInput(
+          { txId: input.hash, index: input.index },
+          false,
+          SELECT_OUTPUTS_TIMEOUT
+        );
+      } catch {
+        // Best-effort: continue releasing remaining UTXOs
+      }
```
Comment (tuliomir) [inline src/new/sendTransaction.ts]: question(non-blocing): Shouldn't we at least emit a debug message in this case? What are the expected situations where this method will fail?

File: src/wallet/sendTransactionWalletService.ts [hunk 3]
```diff
@@ -938,6 +938,18 @@ class SendTransactionWalletService extends EventEmitter implements ISendTransact
...
+  /**
+   * Release all UTXOs that were marked as selected for this transaction.
+   * No-op: wallet-service manages UTXO state server-side.
+   *
+   * @memberof SendTransactionWalletService
+   * @inner
+   */
+  // eslint-disable-next-line class-methods-use-this
+  async releaseUtxos(): Promise<void> {
+    // No-op: wallet-service manages UTXO state server-side
```
Comment (tuliomir) [inline src/wallet/sendTransactionWalletService.ts:950]: question: Is there a plan to implement this feature for the Wallet Service too? An API call to release the stuck UTXOs?

Comment (raul-oliveira) [reply src/wallet/sendTransactionWalletService.ts:950]: Probably not, because the way wallet service locks the utxos is by binding them to a tx proposal. In this use case that we are dealing, we never create a tx proposal before the user approves

File: src/new/sendTransaction.ts (hunk 2 above)
Comment (raul-oliveira) [reply src/new/sendTransaction.ts]: nice catch

Comment (raul-oliveira) [reply src/new/sendTransaction.ts]: done in https://github.com/HathorNetwork/hathor-wallet-lib/pull/1050/commits/de8976f99da5897b0f1602946ce489829f6aad1a

File: __tests__/integration/fullnode-specific/nano_utxo_lock_unlock.test.ts (hunk 1 above)
Comment (raul-oliveira) [reply __tests__/integration/fullnode-specific/nano_utxo_lock_unlock.test.ts:21]: done in https://github.com/HathorNetwork/hathor-wallet-lib/pull/1050/commits/de8976f99da5897b0f1602946ce489829f6aad1a

File: __tests__/integration/helpers/wallet.helper.ts [hunk 4]
```diff
@@ -455,7 +455,7 @@ export async function waitNextBlock(storage) {
 export async function waitTxConfirmed(
   hWallet: HathorWallet,
   txId: string,
-  timeout: number | null | undefined
+  timeout?: number | null
```
Comment (tuliomir) [inline __tests__/integration/helpers/wallet.helper.ts]: suggestion(non-blocking): Reinforcing a suggestion from CodeRabbit, remove `null` from the parameter types, as it's not conceptually correct to pass this value here.

**Review by tuliomir — DISMISSED** [review]

File: src/new/sendTransaction.ts [hunk 5]
```diff
@@ -687,6 +687,33 @@ export default class SendTransaction extends EventEmitter implements ISendTransa
...
+    if (!this.storage) {
+      return;
+    }
+
+    for (const input of this.transaction.inputs) {
+      try {
+        await this.storage.utxoSelectAsInput(
+          { txId: input.hash, index: input.index },
+          false,
+          SELECT_OUTPUTS_TIMEOUT
```
Comment (pedroferreira1) [inline src/new/sendTransaction.ts]: This is not a bug, it just has no use the timeout if we are setting to `false`

**Review by pedroferreira1 — DISMISSED** [review]

File: src/new/sendTransaction.ts (hunk 5 above)
Comment (raul-oliveira) [reply src/new/sendTransaction.ts]: nice catch

Comment (raul-oliveira) [reply src/new/sendTransaction.ts]: done

**Review by tuliomir — APPROVED** [review]

**Review by pedroferreira1 — APPROVED** [review]
